docs/proposals/release-notes.md

```diff
@@ -0,0 +1,194 @@
+as PRs are submitted to the repository.
+
+This direct association of release notes to PRs captures the intention of
+release visibility of the PR at the point an idea is submitted upstream.
+The release notes can then be more easily collected and published when the
+release is ready.
+
+## Background
+
+### The Problem
+
+Release notes are often an afterthought and clarifying and finalizing them
+is often left until the very last minute at the time the release is made.
+This is usually long after the feature or bug fix was added and is no longer on
+the mind of the author.  Worse, the collecting and summarizing of the
+release is often left to those who may know little or nothing about these
+individual changes!
+
+Writing and editing release notes at the end of the cycle can be a rushed,
+interrupt-driven and often stressful process resulting in incomplete,
+inconsistent release notes often with errors and omissions.
+
+### The (general) Solution
+
+Like most things in the development/release pipeline, the earlier you do it,
+the easier it is for everyone and the better the outcome.  Gather your release
+notes earlier in the development cycle, at the time the features and fixes are
+added.
+
+This concept was first implemented at Google for the Borg team to
+capture release notes at submission time.  Since then the concept has
+been adopted widely and used successfully across Google.
+
+#### Then why not just list *every* change that was submitted, CHANGELOG-style?
+
+On larger projects like Kubernetes, showing every single change (PR) would mean
+hundreds of entries.  The goal is to highlight the major changes for a release.
+
+## Options
+
+1. Use of pre-commit and other local git hooks
+   * Experiments here using `prepare-commit-msg` and `commit-msg` git hook files
+     were promising but less than optimal due to the fact that they would
+     require input/confirmation with each commit and there may be multiple
+     commits in a push and eventual PR.
+1. Use of [github templates](https://github.com/blog/2111-issue-and-pull-request-templates)
+   * Templates provide a great way to pre-fill PR comments, but there are no
+     server-side hooks available to parse and/or easily check the contents of
+     those templates to ensure that checkboxes were checked or forms were filled
+     in.
+1. Use of labels enforced by mungers/bots
+   * We already make great use of mungers/bots to manage labels on PRs and it
+     fits very nicely in the existing workflow
+
+## Collection Design
+
+The munger/bot option fits most cleanly into the existing workflow.
+
+The design will include:
+
+1. New labels added to github: `release-note-NONE`, maybe others for new release note categories - see Layout section below
+1. A [new munger](https://github.com/kubernetes/kubernetes/issues/23409) that will:
+  * Initiate a `needs-release-note` label on all new PRs
+  * Disallow/Remove the `lgtm` label on all PRs labeled as `needs-release-note`
```

Comment:
I'm worried we have lots of reviewers who look for assigned issues without LGTM but not many who would filter on do-not-merge in this case. The author of the PR, who would likely see the do-no-merge label would often be unable to set the correct rel-not label (not all PR authors have label/commit privs). But we know that anyone who could set the LGTM will also have to power to set/handle the rel-note issues. So I think the rel-note labeling should be enforced by the reviewer.